server.cpp

```diff
@@ -0,0 +1,96 @@
+#include <iostream>
+#include <string>
+#include <boost/bind.hpp>
+#include <boost/shared_ptr.hpp>
+#include <boost/enable_shared_from_this.hpp>
+#include <boost/asio.hpp>
+
+using boost::asio::ip::tcp;
+
+namespace {
+  const int PORT = 8118;
+}
+
+// shared_ptr and enable_shared_from_this keeps the connection alive
+// as long as there's an operation referring to it.
+class tcp_connection
+    :  public boost::enable_shared_from_this<tcp_connection> {
+  public:
+  	typedef boost::shared_ptr<tcp_connection> pointer;
+
+  	static pointer create(boost::asio::io_service& io_service) {
+  	  return pointer(new tcp_connection(io_service));
+  	}
+
+  	tcp::socket& socket() {
+  	  return socket_;
+  	}
+
+    // async_write() serves data to the client
```

Comment:
i think the comment is for the right function, but the indentation is actually messed up.